libminifi/src/processors/ExtractText.cpp

```diff
@@ -74,11 +111,13 @@ void ExtractText::onTrigger(core::ProcessContext *context, core::ProcessSession
 int64_t ExtractText::ReadCallback::process(std::shared_ptr<io::BaseStream> stream) {
   int64_t ret = 0;
   uint64_t read_size = 0;
+  bool regex_mode = RegexMode.getDefaultValue();
```

Comment:
Why assign to the default value, just because you want to assign it to something?

Reply:
Because getProperty call doesn't assign value to the reference in case the property is not found. 
Yes, it should always be found as these properties have default values, but I prefer not to leave such traps in the  code. 

I was wondering on introducing a function call with a fallback value in signature, like:
```
T TryGetProperty(const std::string name, const T& fallBackValue);
```

Which would make such codes a bit nicer.

Reply:
What trap are you referring to with using default values when properties aren't specified via configuration?

Reply:
Maybe the fact I'm a bit afraid of is the scenario when the property is not required and neither have a default value, so getProperty might return no value. 
In this case it can't occur unless someone removes the defaults, so I removed them here.